configobject/history/history.go

```diff
@@ -38,13 +49,41 @@ func logHistoryCounters() {
 
 	for {
 		<-every20s.C
-		for _, historyType := range [6]string{"state", "notification", "usernotification", "downtime", "comment", "flapping"} {
-			if historyCounter[historyType] > 0 {
-				log.Infof("Added %d %s history entries in the last 20 seconds", historyCounter[historyType], historyType)
-				historyCounterLock.Lock()
-				historyCounter[historyType] = 0
-				historyCounterLock.Unlock()
-			}
+		if historyCounter.state > 0 {
+			log.Infof("Added %d state history entries in the last 20 seconds", historyCounter.state)
+			historyCounterLock.Lock()
+			historyCounter.state = 0
+			historyCounterLock.Unlock()
+		}
+		if historyCounter.notification > 0 {
+			log.Infof("Added %d notification history entries in the last 20 seconds", historyCounter.notification)
+			historyCounterLock.Lock()
+			historyCounter.notification = 0
+			historyCounterLock.Unlock()
+		}
+		if historyCounter.usernotification > 0 {
+			log.Infof("Added %d usernotification history entries in the last 20 seconds", historyCounter.usernotification)
+			historyCounterLock.Lock()
+			historyCounter.usernotification = 0
+			historyCounterLock.Unlock()
+		}
+		if historyCounter.downtime > 0 {
+			log.Infof("Added %d downtime history entries in the last 20 seconds", historyCounter.downtime)
+			historyCounterLock.Lock()
+			historyCounter.downtime = 0
+			historyCounterLock.Unlock()
+		}
+		if historyCounter.comment > 0 {
+			log.Infof("Added %d comment history entries in the last 20 seconds", historyCounter.comment)
+			historyCounterLock.Lock()
+			historyCounter.comment = 0
+			historyCounterLock.Unlock()
+		}
+		if historyCounter.flapping > 0 {
+			log.Infof("Added %d flapping history entries in the last 20 seconds", historyCounter.flapping)
+			historyCounterLock.Lock()
+			historyCounter.flapping = 0
+			historyCounterLock.Unlock()
```

Comment:
This one does almost the same thing six times. Create a new function and call it six times instead.